### DISCUSSION TITLE
How to correctly configure language fallback.

### DISCUSSION THREAD

**Discussion by augustomallmann** [i18next/i18next#1791] [Q&A]
Hi. I've a website with 3 languages and I'm struggling to configure the `/pt-br/` language. The url slug should be `pt-br` and not only `pt`. I was reading [this post](https://github.com/i18next/i18next/issues/1467) but couldn't configure it correctly. The `pt-br` content keeps pulling the `en` (default).

##My Goal:
`my-website.com/pt-br/blog`

##Configuration:

``` 
const languages = ['en', 'pt-BR', 'es'];
const defaultLanguage = 'en';

i18next.use(LanguageDetector).init({
    resources: locales,
    ns: namespaces,
    defaultNS: defaultNamespace,
    returnObjects: true,
    supportedLngs: languages,
    lowerCaseLng: true,
    debug: true,
    nonExplicitSupportedLngs: true,
});
```

##My problem:
Whenever I open a `pt-br` page, it returns the english content (default). What should I do to make it work with `pt-br` and not only `pt`?

**Comment by Ann2827** [comment]
@augustomallmann 
Hi. I found a temporary solution (hack).
```
import en from './en/translation.json';
import pt from './pt/translation.json';
import es from './es/translation.json';

const languages = ['en', 'pt-BR', 'es'];
const defaultLanguage = 'en';

i18next.use(LanguageDetector).init({
    ns: namespaces,
    defaultNS: defaultNamespace,
    returnObjects: true,
    supportedLngs: [...languages, 'pt'],
    lowerCaseLng: true,
    debug: true,
    nonExplicitSupportedLngs: true,
    resources: {
      en: {
        translation: en,
      },
      'pt-BR': {
        translation: pt,
      },
      pt: {
        translation: pt,
      },
      es: {
        translation: es,
      }
    }
});
```
i18next still can't find pt-BR, but it switches to pt (language part). The current language tag will be pt-BR.

There is a second solution. 1 works with pt_BR (with underlining).

**Comment by jamuhl** [comment]
using lowercaselng you should lowercase the lng in your resources too

**Reply by Ann2827** [reply]
My frontend depends on a third-party backend and third-party frontend libraries. They all use the pt-BR format. It is not convenient to convert pt-BR to pt_BR and back all the time

**Reply by jamuhl** [reply]
then don‘t use lowercase option because it lowercase the detected locale and uses that
